packages/import-handler/src/index.ts

```diff
@@ -86,26 +110,94 @@ const sendImportCompletedEmail = async (
     subject: 'Your Omnivore import has completed processing',
-    body: `${urlsEnqueued} URLs have been pcoessed and should be available in your library. ${urlsFailed} URLs failed to be parsed.`,
+    body: `${urlsEnqueued} URLs have been processed and should be available in your library. ${urlsFailed} URLs failed to be parsed.`,
   })
 }
 
 const handlerForFile = (name: string): importHandlerFunc | undefined => {
   const fileName = path.parse(name).name
   if (fileName.startsWith('MATTER')) {
-    return importMatterHistory
+    return importMatterArchive
   } else if (fileName.startsWith('URL_LIST')) {
     return importCsv
   }
 
   return undefined
 }
 
-export const importHandler: EventFunction = async (event, context) => {
-  const data = event as StorageEventData
-  const ctx = context as CloudFunctionsContext
+const urlHandler = async (ctx: ImportContext, url: URL): Promise<void> => {
+  try {
+    // Imports are stored in the format imports/<user id>/<type>-<uuid>.csv
+    const result = await importURL(ctx.userId, url, 'csv-importer')
+    if (result) {
+      ctx.countImported += 1
+    }
+  } catch (err) {
+    console.log('error importing url', err)
+  }
+}
+
+const sendSavePageMutation = async (userId: string, input: unknown) => {
+  const JWT_SECRET = process.env.JWT_SECRET
+  const REST_BACKEND_ENDPOINT = process.env.REST_BACKEND_ENDPOINT
 
-  if (shouldHandle(data, ctx)) {
+  if (!JWT_SECRET || !REST_BACKEND_ENDPOINT) {
+    throw 'Environment not configured correctly'
+  }
+
+  const data = JSON.stringify({
+    query: `mutation SavePage ($input: SavePageInput!){
+          savePage(input:$input){
+            ... on SaveSuccess{
+              url
+              clientRequestId
+            }
+            ... on SaveError{
+                errorCodes
+            }
+          }
+    }`,
+    variables: {
+      input: Object.assign({}, input, { source: 'puppeteer-parse' }),
+    },
+  })
+
+  const auth = (await signToken({ uid: userId }, JWT_SECRET)) as string
+  const response = await axios.post(`${REST_BACKEND_ENDPOINT}/graphql`, data, {
+    headers: {
+      Cookie: `auth=${auth};`,
+      'Content-Type': 'application/json',
+    },
+  })
+  console.log('save page response: ', response)
+
+  /* eslint-disable @typescript-eslint/no-unsafe-member-access */
+  return !!response.data.data.savePage
+}
+
+const contentHandler = async (
+  ctx: ImportContext,
+  url: URL,
+  title: string,
+  originalContent: string,
+  parseResult: Readability.ParseResult
+): Promise<void> => {
+  const requestId = uuid()
+  const apiResponse = await sendSavePageMutation(ctx.userId, {
+    url,
+    clientRequestId: requestId,
+    title,
+    originalContent,
+    parseResult,
+  })
+  if (!apiResponse) {
+    return Promise.reject()
+  }
+  return Promise.resolve()
+}
+
+const handleEvent = async (data: StorageEvent) => {
+  if (shouldHandle(data)) {
     console.log('handling csv data', data)
 
     const stream = storage
```

Comment:
We could create read stream after the below checking

Reply:
nice one